controllers/storagecluster/routes.go

```diff
@@ -0,0 +1,154 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+
+	routev1 "github.com/openshift/api/route/v1"
+	ocsv1 "github.com/openshift/ocs-operator/api/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type ocsCephRGWRoutes struct{}
+
+// ensureCreated ensures that CephObjectStore resources exist in the desired
+// state.
+func (obj *ocsCephRGWRoutes) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	reconcileStrategy := ReconcileStrategy(instance.Spec.ManagedResources.CephObjectStores.ReconcileStrategy)
+	if reconcileStrategy == ReconcileStrategyIgnore {
+		return nil
+	}
+	platform, err := r.platform.GetPlatform(r.Client)
+	if err != nil {
+		return err
+	}
+	if avoidObjectStore(platform) {
+		r.Log.Info(fmt.Sprintf("not creating a Ceph RGW route because the platform is '%s'", platform))
+		return nil
+	}
+
+	ocsCephRoutes, err := r.newCephRGWRoutes(instance)
+	if err != nil {
+		return err
+	}
+	err = r.createCephRGWRoutes(ocsCephRoutes, instance)
+	if err != nil {
+		r.Log.Error(err, "could not create Routes")
+		return err
+	}
+
+	return nil
+}
+
+// ensureDeleted deletes the CephObjectStores owned by the StorageCluster
+func (obj *ocsCephRGWRoutes) ensureDeleted(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) error {
+	foundRoute := &routev1.Route{}
+	routes, err := r.newCephRGWRoutes(sc)
+	if err != nil {
+		return err
+	}
+
+	for _, route := range routes {
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: route.Name, Namespace: sc.Namespace}, foundRoute)
```

Comment:
Instead of `Get()`, `Delete()`, `Get()`, you could just `Delete()` and then ignore the error if `IsNotFound(err)`

Reply:
This is the general pattern we're following in other files. The main thing this avoid is sending a `Delete()` is the deletion timestamp is not zero. But, conversely, we are wasting time on a `Get()` every time the deletion timestamp is zero, so it may not be any better. Also, we probably don't gain anything from distinguishing between "Found and not marked for deletion" and "Found and marked for deletion", since we are already committed to deleting by the time we get to this function.

Might not be a bad idea to refactor the other `ensureDeleted()` functions in a later PR.

Reply:
This would be better done in a separate PR to change the behaviour in all the files.